src/Microsoft.Data/DataFrame.cs

```diff
@@ -9,6 +9,14 @@
 
 namespace Microsoft.Data
 {
+    public enum JoinAlgorithm
+    {
+        Left,
+        Right,
+        Outer,
```

Comment:
This might be my old T-SQL experience hindering me, but "Left" and "Right" are also "Outer" joins, so having an "Outer" option confuses me. Would it be better if we called this `FullOuter`?

Reply:
Sure. I just went with "Outer" because Pandas calls it that